src/lib/action.ts

```diff
@@ -84,6 +84,18 @@ export const actionName = (action: Action): string => {
   return name[0].toUpperCase() + name.slice(1);
 };
 
+class LaTeXError extends MalformedExpressionException {
+  constructor(message: string, sourceTeX?: string) {
+    super(
+      sourceTeX === undefined
+        ? message
+        : `${message}
+
+in LaTeX ${sourceTeX}`
```

Comment:
nit: should have a period at the end of the error message?

Reply:
Hmm, I generally avoid this, especially when the error message goes to the developer and not the end user (this is mainly the case here). Additionally, this has an insertion, and adding a period to the end would have you believe that the period was actually included in the LaTeX source. (n.b., anyone can already see the entered LaTeX by looking at attributes of child elements of the logged SVG element, but this is added for convencience, mainly so that users can copy the entered source instead of retyping it [adding automatic copying would've been too intrusive of a change], but also so that they can immediately see what exactly they have typed. Adding the period hurts on both fronts.) If you look in your console, you'll likely see errors of the form `Failed to load resource: net::ERR_BLOCKED_BY_CLIENT`, which have quite similar structure to this one.

Of course, it's a different situation with https://github.com/cjquines/qboard/blob/e776219abd8bcfcc8242010b7d6249daa045e1e7/src/lib/action.ts#L252-L256 where neither concern applies; in that case we of course add the period, although even in this case doing so is contested by many, especially as it's not a complete sentence (a third reason for not doing it in the error class).